esi_leap/api/controllers/v1/offer.py

```diff
@@ -195,9 +197,30 @@ def delete(self, offer_id):
         if len(permission) == 0:
             raise exception.OfferNotFound(offer_id=offer_id)
 
-        OffersController._verify_resource_permission(cdict, o_id[0].to_dict())
+        OffersController._verify_resource_permission(cdict,
+                                                     o_objects[0].to_dict())
 
-        o_id[0].cancel()
+        o_objects[0].cancel()
+
+    @staticmethod
+    def _get_offer_by_uuid_or_name(uuid_or_name, cdict):
```

Comment:
Ah, here's the duplication I'm talking about. I understand the logic is slightly different, but that can be solved with a flag to the function so that it covers what you need here and with contracts.

Reply:
I can solve this with a flag that just checks if you need to do contract creation or offer management, but it seems like a very sloppy way to do so. But I otherwise don't know how to combine them.

Reply:
The reason there are two functions that do similar but not quite identical things is because of the fact that resource management is poorly defined for objects of expired/cancelled status. The issue is further compounded by the name conflicts issue. I can do the resolution with a flag but then we have some pretty obfuscated functions.

Reply:
Can you break them up into three util functions for now then (and write tests)? After that, it may become clearer whether it's worth combining them or not.

Reply:
yeah I think they can be combined in a smarter way. Looking at some of the other code I realize it could be written a bit better.